astroid/brain/brain_typing.py

```diff
@@ -75,14 +75,18 @@ def _looks_like_typing_subscript(node):
 def infer_typing_attr(node, context=None):
     """Infer a typing.X[...] subscript"""
     try:
-        value = next(node.value.infer())
+        for value in node.value.infer():
```

Comment:
Not sure why we were not passing the `context` parameter to `value` inference. Can you add it here and see if it has any unintended side effects? (e.g. if certain tests start to fail, let's keep it removed)